barman/postgres.py

```diff
@@ -1235,11 +1235,15 @@ def stop_exclusive_backup(self):
             if self.server_version >= 150000:
                 raise PostgresObsoleteFeature("15")
 
+            # Call pg_stop_backup with TRUE, FALSE.
+            # TRUE indicates exclusive backup and FALSE tells PostgreSQL we do not
+            # want to wait for the required WALs to be archived. This is because
+            # barman will do its own waiting at a later stage of the backup process.
             cur.execute(
                 "SELECT location, "
                 "({pg_walfile_name_offset}(location)).*, "
                 "now() AS timestamp "
-                "FROM {pg_backup_stop}() AS location".format(**self.name_map)
+                "FROM {pg_backup_stop}(TRUE, FALSE) AS location".format(**self.name_map)
```

Comment:
The (bool, bool) form of pg_stop_backup returns a record: `record ( lsn pg_lsn, labelfile text, spcmapfile text )` instead of the pg_lsn which the () form returns. We therefore need to extract the pg_lsn in this query to avoid breaking exclusive backup.

Reply:
Also, double check that the permissions in the docs and the permissions check are still correct since as of this commit we will no longer be using the `()` form of `pg_stop_backup`.